platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DefaultConfigurationCacheDegradationController.kt

```diff
@@ -25,21 +25,37 @@ import org.gradle.execution.plan.TaskNode
 import org.gradle.internal.cc.impl.services.DeferredRootBuildGradle
 import org.gradle.internal.service.scopes.Scope.BuildTree
 import org.gradle.internal.service.scopes.ServiceScope
-import java.util.Collections
+import org.gradle.vcs.internal.VcsMappingsStore
 import java.util.concurrent.ConcurrentHashMap
 
 @ServiceScope(BuildTree::class)
 internal class DefaultConfigurationCacheDegradationController(
     private val deferredRootBuildGradle: DeferredRootBuildGradle,
     private val configurationTimeBarrier: ConfigurationTimeBarrier,
+    private val vcsMappingsStore: VcsMappingsStore
 ) : ConfigurationCacheDegradationController {
 
     private val logger = Logging.getLogger(DefaultConfigurationCacheDegradationController::class.java)
     private val tasksDegradationRequests = ConcurrentHashMap<Task, List<Provider<String>>>()
-    private val collectedDegradationReasons = mutableMapOf<Task, List<String>>()
+    private val collectedDegradationReasons = mutableMapOf<DegradationContext, List<String>>()
 
-    val degradationReasons: Map<Task, List<String>>
-        get() = Collections.unmodifiableMap(collectedDegradationReasons)
+    val hasDegradationReasons: Boolean by lazy { collectedDegradationReasons.isNotEmpty() }
+
+    val hasTaskDegradationReasons: Boolean by lazy {
+        collectedDegradationReasons.any {
+            e -> e.key is DegradationContext.Task
```

Comment:
🤔 I wonder why it's `e` here and below?

Reply:
I sometimes use the initial for the kind of object we are consuming ("e" for entry, "t" for task, "p" for problem).